unlock-app/src/pages/_app.js

```diff
@@ -13,13 +13,18 @@ import { ConfigContext } from '../utils/withConfig'
 import web3Middleware from '../middlewares/web3Middleware'
 import currencyConversionMiddleware from '../middlewares/currencyConversionMiddleware'
 import storageMiddleware from '../middlewares/storageMiddleware'
+import walletMiddleware from '../middlewares/walletMiddleware'
 
 const config = configure()
 
 const __NEXT_REDUX_STORE__ = '__NEXT_REDUX_STORE__'
 
 function getOrCreateStore(initialState, history) {
-  const middlewares = [web3Middleware, currencyConversionMiddleware]
+  const middlewares = [
+    web3Middleware,
+    currencyConversionMiddleware,
+    walletMiddleware,
```

Comment:
can I assume that a future PR will recognize context and load only the walletMiddleware for the paywall route?

Reply:
Not sure what you mean here?
At this point the `_app` will load all middlewares at all points because it does not know which route the user is currently seeing/might see? 
We also need the `walletMiddleware` for both the paywall and daashboard because this is the piece that interracts with the user wallet to send transactions (create lock or purchase key for example)

Reply:
In other (unrelated) PRs I am working on, I created a middleware that should only be loaded on paywall.js to do the inter-window communication. Since the web3Middleware won't be needed on the paywall, at some point, we can use this idea to not load it if the route is `'/paywall'`. That's what I was alluding to.

Reply:
Actually we need the web3Middleware on the paywall.